starter_code/hangman.js

```diff
@@ -1,44 +1,57 @@
 let hangman;
-
-// class Hangman {
-//   constructor() {
-
-//   }
-
-//   getWord() {
-
-//   }
-
-//   checkIfLetter(keyCode) {
-
-//   }
-
-//   checkClickedLetters(key) {
-
-//   }
-
-//   addCorrectLetter(i) {
-
-//   }
-
-//   addWrongLetter(letter) {
-
-//   }
-
-//   checkGameOver() {
-
-//   }
-
-//   checkWinner() {
-
-//   }
-
-// }
+class Hangman {
+ constructor() {
+ this.words = ["lentes" ,"carro " ,"jazz", "limon" ];
+ this.secretWord = "";
+ this.letters = [];
+ this.guessedLetter = "";
+ this.errorsLeft = 10;
+ this.letters
+ }
+  getWord() {
+    let chosenWord = this.words[Math.floor(Math.random()*this.words.length)];
+    return chosenWord;
+  }
+
+  checkIfLetter(keyCode) {
+    return (keyCode >= 65 && keyCode <= 90);
+  }
+
+  checkClickedLetters(key) {
+    if(this.letters.indexOf(key) != -1) {
+      return false;
+    }
+    return true;
+  }
+
+  checkWinner() {
+    if(this.guessedLetter.length == this.secretWord.length){
+      return true;
+    }
+
+    return false;
+  }
+
+  addCorrectLetter(letter) {
+    this.guessedLetter += "letter";
```

Comment:
one solution for this is: `this.guessedLetter += this.secretWord.charAt(letter).toUpperCase();`